### PR TITLE
Guide for Testing with JUnit5

### PR DESCRIPTION
**Pull request by ljamen**
Now ready for first review.

### CODE REVIEW

**Comment by ljamen** [comment]
I did not want to duplicate what is already in the MP testing doc. This guide assumes the user has reviewed that doc and is ready for the 'hands-on' examples.

There may be other resources for the Summary. I added a couple that I thought could be useful.

**Comment by ljamen** [comment]
From the blog article: A small bug crawled into our code base, and as a result the artifact needed for this article did not get deployed to maven central as part of 2.1.0 release. We are working on a fix!

Is this something we need to address or fix in this guide/doc?

File: docs/mp/guides/38_testing_junit5.adoc [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+=== Inject a WebTarget
+
+The test is only useful if it invokes the server and verifies the result. To support testing, you can inject a `WebTarget` that is configured for the currently running server (it can also be a parameter to a test method). We can use the target to invoke our endpoint and validate the result.
+
+
+[source,java]
+.Updated Class with webTarget
+----
+import static org.junit.jupiter.api.Assertions.assertEquals;@HelidonTest
```
Comment (dalexandrov) [inline docs/mp/guides/38_testing_junit5.adoc:96]: Please separate lines after ";"
@HelidonTest should be right above the class.

File: docs/mp/guides/38_testing_junit5.adoc [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@HelidonTest
+@DisableDiscovery
+@AddExtension(ConfigCdiExtension.class)
+@AddBean(GreetTest.ConfiguredBean.class)
+@AddConfig(key = "test.message", value = "Hello Blog!")
```
Comment (dalexandrov) [inline docs/mp/guides/38_testing_junit5.adoc:230]: Maybe "Hello Guide!" ?

**Review by dalexandrov — DISMISSED** [review]

File: docs/mp/guides/38_testing_junit5.adoc [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+=== Disable Discovery
+
+If you want to disable discovery and only add custom extensions and beans, then use the following annotation:
+
+[source, java]
+----
+@DisableDiscovery
+----
+
+NOTE: This annotation is typically used in conjunction with `@AddBeans` and/or `@AddExtension`. As you have seen in standard test output, Helidon starts with the full MicroProfile features enabled.
```
Comment (tomas-langer) [inline docs/mp/guides/38_testing_junit5.adoc:205]: maybe add "By default Helidon starts with dependencies defined in `pom.xml`"

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: docs/mp/guides/38_testing_junit5.adoc [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+[source,xml]
+.Maven dependency
+----
+<dependency>
+    <groupId>io.helidon.microprofile.tests</groupId>
+    <artifactId>helidon-microprofile-tests-junit5</artifactId>
+    <scope>test</scope>
+</dependency>
+----
```
[the remark this answers is not in the document]
Comment (ljamen) [reply docs/mp/guides/38_testing_junit5.adoc:51]: I don't know if I have documented/formatted this correctly. @dalexandrov Can you please check?

File: docs/mp/guides/38_testing_junit5.adoc [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+include::{common-page-prefix-inc}[tag=common-prereqs]
+
+
+== Dependencies
+To start using this feature, add the following dependencies to the testing module:
+
+[source,xml]
+.Maven dependencies
+----
+<dependencies>
```
Comment (ljamen) [inline docs/mp/guides/38_testing_junit5.adoc:39]: @Tomas-Kraus I added the additional dependency for junit jupiter but I'm not sure if I used the correct version (5.5.2) or formatting.

Comment (tomas-langer) [reply docs/mp/guides/38_testing_junit5.adoc:39]: The version does not need to be defined (when using our application parent poms - it is managed by Helidon)

**Review by tomas-langer — COMMENTED** [review]
Except for the version of JUnit, which I think does not need to be defined, the guide is really nice! Also specifying a version would require constant updates to the guide when a version is changed (which is easy to be missed).

In current version of Helidon, we use JUnit version 5.7.0.

File: docs/mp/guides/38_testing_junit5.adoc (hunk 5 above)
Comment (ljamen) [reply docs/mp/guides/38_testing_junit5.adoc:39]: removed version from dependencies

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**